### PR TITLE
Tests for sycl::get_kernel_id and sycl::get_kernel_ids

### PR DESCRIPTION
**Pull request by vasilytric**
Some of these test required containing in separate *.exe-file: `get_kernel_ids_multiple_kernels`, `get_kernel_ids_single_kernel` and `get_kernel_ids_no_kernels` because these files required stricted list of available kernels
for build this tests need call building process for each test separately: `make test_get_kernel_ids_multiple_kernels`, `make test_get_kernel_ids_single_kernel` and `make test_get_kernel_ids_no_kernels`
But if use `make` all tests will be builded

### CODE REVIEW

File: tests/kernel_bundle/get_kernel_id_function.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    set_test_info(out, TOSTRING(TEST_NAME), TEST_FILE);
+  }
+
+  /** execute the test
+   */
+  void run(util::logger &log) override {
+    /** Check return type of sycl::get_kernel_id()
+     */
+    run_verification<by_queue>(log, [&](sycl::queue &queue) {
+      using kernel_name = class k_id_ret_type;
```
Comment (keryell) [inline tests/kernel_bundle/get_kernel_id_function.cpp:34]: Interesting, I have never thought to this syntax.

**Review by keryell — APPROVED** [review]

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -1,19 +1,31 @@
+set(tests_dir ${CMAKE_CURRENT_SOURCE_DIR})
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 
-if(SYCL_CTS_TEST_FILTER)
```
Comment (ProGTX) [inline tests/CMakeLists.txt:3]: Why these changes? They seem to go the other direction of what @psalz 's PRs did.

File: tests/CMakeLists.txt [hunk 3]
```diff
@@ -182,12 +181,37 @@ function(add_cts_test)
...
+  foreach(ind_test IN LISTS tests_list)
+    if(EXISTS "${ind_test}")
+      get_filename_component(cpp_name "${ind_test}" NAME_WE)
+      set(test_exe_name "${cpp_name}")
+      set(test_cases_list "${ind_test}")
+
+      add_cts_test_helper(${test_exe_name} "${test_cases_list}")
+    else()
+      # this comment will be removed after testing TMT run
+      message(STATUS ">>>>>>>>>>>>>>>>>>>>> no file named ${ind_test}")
```
Comment (ProGTX) [inline tests/CMakeLists.txt:208]:
```suggestion
      # ...
```
There should be an explanation for why it's OK to skip a file that doesn't exist

File: tests/CMakeLists.txt [hunk 4]
```diff
@@ -182,12 +181,37 @@ function(add_cts_test)
...
+  if (tests_list)
+    get_filename_component(test_dir ${CMAKE_CURRENT_SOURCE_DIR} NAME)
+    set(test_exe_name ${test_dir})
+    set(test_cases_list "${ARGN}")
+
+    add_cts_test_helper(${test_exe_name} "${test_cases_list}")
   endif()
+endfunction()
+
+function(add_independent_cts_tests)
```
Comment (ProGTX) [inline tests/CMakeLists.txt:197]: Can you please add some comments to this function, what's the purpose of it. Additionally, please add comments to `add_cts_test` - I think the difference between them needs to be clearly communicated.

**Review by ProGTX — CHANGES_REQUESTED** [review]

File: tests/CMakeLists.txt (hunk 4 above)
Comment (vasilytric) [reply tests/CMakeLists.txt:199]: comments was added in this commit: 78cae14b03d7ce5f688f024424fb5b91a0171a76

File: tests/CMakeLists.txt (hunk 3 above)
Comment (vasilytric) [reply tests/CMakeLists.txt]: now, generating CMake files process will be stopped, if at least one file doesn't exist
I'll do it in this commit: 21f8fc9e15876fd4e9c2de23deb84e3327021dd0

File: tests/CMakeLists.txt (hunk 2 above)
Comment (vasilytric) [reply tests/CMakeLists.txt]: Whoops, these changes are from the older code base
I've reverted it in this commit: f18f47ca70a3ec5125fa8fb0046431d373de898f

**Review by keryell — APPROVED** [review]

**Comment by bader** [comment]
@vasilytric, CI says that computecpp and hipsycl doesn't support these features yet.
Please, update corresponding filters - https://github.com/KhronosGroup/SYCL-CTS/tree/SYCL-2020/ci.

**Comment by vasilytric** [comment]
> @vasilytric, CI says that computecpp and hipsycl doesn't support these features yet. Please, update corresponding filters - https://github.com/KhronosGroup/SYCL-CTS/tree/SYCL-2020/ci.

@bader Oh, sorry. these files was updated in this commit: b4692acefc2039c4ab084fa06485d60b8df12839

**Comment by psalz** [comment]
You say that these translation units must not be linked together -- doesn't this mean that the `test_all` executable will never pass? Should these tests be excluded from `test_all`? Of course then it no longer *tests all*...

**Comment by bader** [comment]
@vasilytric, please, resolve merge conflicts.

**Comment by vasilytric** [comment]
> @vasilytric, please, resolve merge conflicts.

@bader Done

**Comment by bader** [comment]
@vasilytric, please, update pull requests depending on changes from this PR to get valid CI checks status.
